src/twisted/python/test/test_systemd.py

```diff
@@ -7,158 +7,163 @@
 
 
 import os
+from typing import Dict, Mapping, Sequence
 
-from twisted.python.systemd import ListenFDs
-from twisted.trial.unittest import TestCase
-
-
-class InheritedDescriptorsMixin:
-    """
-    Mixin for a L{TestCase} subclass which defines test methods for some kind of
-    systemd sd-daemon class.  In particular, it defines tests for a
-    C{inheritedDescriptors} method.
-    """
-
-    def test_inheritedDescriptors(self):
-        """
-        C{inheritedDescriptors} returns a list of integers giving the file
-        descriptors which were inherited from systemd.
-        """
-        sddaemon = self.getDaemon(7, 3)
-        self.assertEqual([7, 8, 9], sddaemon.inheritedDescriptors())
+from hamcrest import assert_that, equal_to, not_
+from hypothesis import given
+from hypothesis.strategies import dictionaries, integers, lists
 
-    def test_repeated(self):
-        """
-        Any subsequent calls to C{inheritedDescriptors} return the same list.
-        """
-        sddaemon = self.getDaemon(7, 3)
-        self.assertEqual(
-            sddaemon.inheritedDescriptors(), sddaemon.inheritedDescriptors()
-        )
+from twisted.python.systemd import ListenFDs
+from twisted.trial.unittest import SynchronousTestCase
+from .strategies import systemd_descriptor_names
 
 
-class MemoryOnlyMixin:
+def initializeEnvironment(count: int, pid: object) -> Dict[str, str]:
     """
-    Mixin for a L{TestCase} subclass which creates creating a fake, in-memory
-    implementation of C{inheritedDescriptors}.  This provides verification that
-    the fake behaves in a compatible way with the real implementation.
+    Create a copy of the process environment and add I{LISTEN_FDS} and
+    I{LISTEN_PID} (the environment variables set by systemd) to it.
```

Comment:
I think the description can be generic. No need to mention the exact env var names.

I know that this was already in trunk... just a minor comment

```suggestion
    Return a copy of the process environment with I{systemd} specific variables added to it.
```